### ISSUE TITLE
Codebuild artifacts packaging changes is not applied

### ISSUE THREAD

**Issue by SiahaanBernard** [terraform-providers/terraform-provider-aws#2050]
Hi
I am creating aws codebuild project with artifact packaging type "NONE", however i want to change this artifact packaging type to "ZIP" but when i change the packaging type to "ZIP", i run terraform plan and there is no changes detected, run terraform apply doesn't make a changes to the codebuild project.

### Terraform Version
v0.10.4

### Affected Resource(s)
- AWS CodeBuild

### Terraform Configuration Files
here is my first terraform configuration
```hcl
resource "aws_codebuild_project" "packaging_test" {
  name         = "packaging_test"
  service_role = "code_build_run"

  "source" {
    type      = "CODECOMMIT"
    location  = "${var.codecommit_url}"
    buildspec = "buildspec_build.yml"
  }

  "environment" {
    compute_type = "BUILD_GENERAL1_SMALL"
    image        = "aws/codebuild/ubuntu-base:14.04"
    type         = "LINUX_CONTAINER"
  }

  "artifacts" {
    type      = "S3"
    location  = "${var.s3_bucket_path}"
    name      = "packaging_test"
    packaging = "NONE"
  }
}
```
then change the artifacts section to
```hcl
...
  "artifacts" {
    type      = "S3"
    location  = "${var.s3_bucket_path}"
    name      = "packaging_test"
    packaging = "ZIP"
  }
...
```

### Expected Behavior
- Codebuild artifact packaging type should be change from NONE become ZIP on AWS

### Actual Behavior
- terraform does not detect any change and does nothing

**Comment by barryoneill** [comment]
I'm also seeing changes in the buildspec file not being picked up (e.g. if I change one of the pre_build commands), so the apply sees nothing to do.   Could be related?  (Terraform v0.11.1)

**Comment by bflad** [comment]
The fixes for having the `aws_codebuild_project` resource `artifacts` configuration block perform better drift detection and updates have been merged and will release with version 2.22.0 of the Terraform AWS Provider, tomorrow. 👍
